forge-gui/res/cardsfolder/upcoming/tasha_the_witch_queen.txt

```diff
@@ -0,0 +1,14 @@
+Name:Tasha, the Witch Queen
+ManaCost:3 U B
+Types:Legendary Planeswalker Tasha
+Loyalty:4
+Text:CARDNAME can be your commander.
+T:Mode$ SpellCast | ValidCard$ Card.OppOwn | ValidActivatingPlayer$ You | Execute$ TrigToken | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast a spell you don't own, create a 3/3 black Demon creature token.
+SVar:TrigToken:DB$ Token | TokenOwner$ You | TokenAmount$ 1 | TokenScript$ b_3_3_demon
+A:AB$ Draw | Cost$ AddCounter<1/LOYALTY> | Planeswalker$ True | Defined$ You | NumCards$ 1 | SubAbility$ DBExileForEach | SpellDescription$ Draw a card. For each opponent, exile up to one target instant or sorcery card from that player's graveyard and put a page counter on it.
+SVar:DBExileForEach:DB$ ChangeZone | Origin$ Graveyard | Destination$ Exile | ValidTgts$ Instant.OppCtrl,Sorcery.OppCtrl | TgtPrompt$ Choose up to one target instant or sorcery card in each opponent's graveyard | TargetMin$ 0 | TargetMax$ OneEach | TargetsWithDifferentControllers$ True | SubAbility$ DBCounter | RememberChanged$ True
+SVar:DBCounter:DB$ PutCounter | CounterType$ PAGE | CounterNum$ 1 | Defined$ Remembered | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:OneEach:PlayerCountOpponents$Amount
+A:AB$ Play | Cost$ SubCounter<3/LOYALTY> | Planeswalker$ True | Ultimate$ True | Valid$ Card.counters_GE1_PAGE | ValidZone$ Exile | WithoutManaCost$ True | ValidSA$ Spell | Amount$ 1 | Controller$ You | Optional$ True | SpellDescription$ You may cast a spell from among cards in exile with page counters on them without paying its mana cost.
```

Comment:
I don't think we usually call abilities that could be activated right away (even though it would be a waste) Ultimate

I think the AI also slightly prefers anything with the Ultimate tag? @Agetian 

We probably want to add some kind of AI check here so the AI doesn't try using this before there is anything valid in Exile

Reply:
Thank you for your reviews so far, I will soon push the recommended changes. 

I think in testing I followed the examples of walkers such as Chandra, Flamecaller and Gideon, Ally of Zendikar, who respectively have instant -X and -4 abilities that are counted in their scripts as ultimates. Is it the philosophy that all walkers must have an "ultimate" within their scripts somewhere, or should we apply "ultimate" in the future only to minus abilities that can't be activated right away?

I'm unfamiliar with adding complex AI checks to scripts. Should I instead remove the ultimate keyword altogether here?

Reply:
Yeah, that's why I hedged with "usually"... it's a bit subjective. You'll definitely also find walkers with a last ability that isn't tagged as an Ultimate. Let's see what some other folks think in the Discord

I don't think the AI check would be too tough in this case... stuff like Idle Thoughts and Vilespawn Spider might get you started. `AICheckSVar$ X | AISVarCompare$ GE1` with
`SVar:X:Count$ValidExile`... I'm sure you can handle it from there, but check Etrata if you get stuck ;)

Reply:
With that AICheckSVar the AI just won't activate it unless that check passes

Reply:
PlayAi should already have some basic sanity check to prevent that anyway

might just not care for other playability factors in this case yet

Reply:
Added AI priority flag to check for page counters on exiled cards before activating -3. Awaiting ruling on whether or not to keep the ultimate flag with the AI check or remove it.

Reply:
Removed ultimate tag from -3 ability